rllib/agents/dreamer/dreamer.py

```diff
@@ -250,3 +278,44 @@ def execution_plan(workers, config, **kwargs):
             )
         )
         return rollouts
+
+    @override(Trainer)
+    def training_iteration(self) -> ResultDict:
```

Comment:
Awesome stuff @amogkam ! The algo looks very clean. Thanks also for your feedback on the new API.